Description: Skip processing fdtdir on qemu-riscv64_smode target
 it crashes the riscv qemu VM.
Author: Dimitri John Ledkov <xnox@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1925267


--- u-boot-2021.01+dfsg.orig/cmd/pxe_utils.c
+++ u-boot-2021.01+dfsg/cmd/pxe_utils.c
@@ -471,6 +471,7 @@ static int label_boot(struct cmd_tbl *cm
 
 		if (label->fdt) {
 			fdtfile = label->fdt;
+#if !(defined(CONFIG_TARGET_QEMU_VIRT) && defined(CONFIG_RISCV_SMODE))
 		} else if (label->fdtdir) {
 			char *f1, *f2, *f3, *f4, *slash;
 
@@ -513,6 +514,7 @@ static int label_boot(struct cmd_tbl *cm
 			snprintf(fdtfilefree, len, "%s%s%s%s%s%s",
 				 label->fdtdir, slash, f1, f2, f3, f4);
 			fdtfile = fdtfilefree;
+#endif
 		}
 
 		if (fdtfile) {
